loki/ir/nodes.py

```diff
@@ -257,17 +257,10 @@ class InternalNode(Node, _InternalNode):
 
     _traversable = ['body']
 
-    @model_validator(mode='before')
+    @field_validator('body', mode='before')
```

Comment:
Likely my ignorance of pydantic showing here: The `pre_init` method did also stuff to `args` and `kwargs` before. Is this implicitly handled now or not required anymore?

Reply:
Yes, but this one worked on the entire "Model" in pydantic speak, so it got a single `ArgsKwargs` argument that held all the constructor arguments as `ArgsKwarg.args` and `ArgsKwargs.kwargs`. This became an issue, as the `ArgsKwargs` object is immutable, but the `ArgsKwarg.kwargs` dict could be changed. This means, if any kw-args were given, I could change the dict, but I cannot replace `None` with `()` as `ArgsKwargs.args = ()` is not allowed. 

Long story short, this new validator is applied to each arg individual when pydantic creates the `ArgsKwargs` object, so all our troubles fade away and pydantic takes care of identifying unnamed args for free.